validphys2/src/validphys/core.py

```diff
@@ -266,83 +264,153 @@ def get_kinlabel_key(process_label):
                          "variables matching  the process %s Check the "
                          "labels defined in commondata.cc. " % (l)) from e
 
-CommonDataMetadata = namedtuple('CommonDataMetadata', ('name', 'nsys', 'ndata', 'process_type'))
 
-def peek_commondata_metadata(commondatafilename):
-    """Check some basic properties commondata object without going though the
-    trouble of processing it on the C++ side"""
-    with open(commondatafilename) as f:
-        try:
-            l = f.readline()
-            name, nsys_str, ndata_str = l.split()
-            l = f.readline()
-            process_type_str = l.split()[1]
-        except Exception:
-            log.error(f"Error processing {commondatafilename}")
-            raise
+class CommonDataSpec(TupleComp):
 
-    return CommonDataMetadata(name, int(nsys_str), int(ndata_str),
-                              get_kinlabel_key(process_type_str))
+    def __init__(self, name, variant, metadata_file=None):
```

Comment:
Do we really need `name` and `metadata_file`? Could we not have just a folder and derive the name from that?